pkg/provider/aws/resources/lb.go

```diff
@@ -0,0 +1,135 @@
+package resources
+
+import (
+	"fmt"
+
+	"github.com/klothoplatform/klotho/pkg/core"
+	"github.com/klothoplatform/klotho/pkg/sanitization/aws"
+)
+
+var (
+	loadBalancerSanitizer = aws.LoadBalancerSanitizer
+	targetGroupSanitizer  = aws.TargetGroupSanitizer
+)
+
+const (
+	LOAD_BALANCER_TYPE            = "load_balancer"
+	TARGET_GROUP_TYPE             = "target_group"
+	LISTENER_TYPE                 = "load_balancer_listener"
+	NLB_INTEGRATION_URI_IAC_VALUE = "nlb_uri"
+)
+
+type (
+	LoadBalancer struct {
+		Name                   string
+		ConstructsRef          []core.AnnotationKey
+		IpAddressType          string
+		LoadBalancerAttributes map[string]string
+		Scheme                 string
+		SecurityGroups         []*SecurityGroup
```

Comment:
well if the security group gets modified we just want to make sure that its in sync everywhere was my thought. Also all of the IRs imlpement the core.Resource interface as pointers so its more natural it feels like. not sure if it would have issues not using pointers though